File: b/Disk.java
enum DiskType{SSD, HDD};

class Disk
{
	private DiskType type;
	private int capacity;
	private int price;

	// construct
	public Disk(DiskType type, int capacity, int price)
	{
		this.type = type;
		this.capacity = capacity;
		this.price = price;
	}

	// default construct
	public Disk(){}

	public void setType(DiskType type) { this.type = type; }
	public void setCapacity(int capacity) { this.capacity = capacity; }
	public void setPrice(int price) { this.price = price; }

	public DiskType getType() { return this.type; }
	public int getCapacity() { return this.capacity; }
	public int getPrice() { return this.price; }

	public void printDisk()
	{
		System.out.println("Tipe Disk : " + this.type);
		System.out.println("Kapasitas Disk : " + this.capacity);
		System.out.println("Harga Disk : " + this.price);
	}
}

File: b/Disk.php
<?php declare(strict_types=1);

// enum is only supported on PHP version 8 >= 8.1.0!
enum DiskType
{
	case SSD;
	case HDD;
}

class Disk
{
	private DiskType $type;
	private int $capacity;
	private int $price;

	public function __construct(DiskType $type, int $capacity, int $price)
	{
		$this->type = $type;
		$this->capacity = $capacity;
		$this->price = $price;
	}

	public function setType(DiskType $type): void { $this->type = $type; }
	public function setCapacity(int $capacity): void { $this->capacity = $capacity; }
	public function setPrice(int $price): void { $this->price = $price; }

	public function getType(): DiskType { return $this->type; }
	public function getCapacity(): int { return $this->capacity; }
	public function getPrice(): int { return $this->price; }

	public function printDisk()
	{
		switch($this->type)
		{
			case DiskType::SSD:
				echo "Tipe Disk : SSD" . "<br>";
				break;
			case DiskType::HDD:
				echo "Tipe Disk : HDD" . "<br>";
		}
		echo "Kapasitas Disk : " . $this->capacity . "<br>";
		echo "Harga Disk : " . $this->price . "<br>";
	}
}

?>